deploy.py

```
@@ -0,0 +1,307 @@
+            lim_mb = lim / (1024 * 1024)
+            logger.info(f"Container memory usage at start: {cur_mb:.1f} MB / {lim_mb:.1f} MB")
+        else:
+            logger.info(f"Container memory usage at start: {cur_mb:.1f} MB (no cgroup limit detected)")
+
+
+
+    # Validate and fetch required API keys early
+    def _require_env(name: str) -> str:
+        val = os.getenv(name)
+        if not val or not val.strip():
+            raise RuntimeError(f"Missing required environment variable: {name}")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)